rxtools/source/lib/fs.c

```diff
@@ -99,22 +99,48 @@ void FileClose(File *Handle)
     f_close(Handle);
 }
 
-int FileCopy(char* dest, char* source){
+int FileCopy(char* dest, char* source)
+{
 	File out;
 	File in;
-	if(!FileOpen(&in, source, 0)) return -1;
-	FileOpen(&out, dest, 1);
-	unsigned int chunk_size = 0x4000;
+	if (!FileOpen(&in, source, 0)) return -1;
+	if (!FileOpen(&out, dest, 1)) return -1;
+	
+	unsigned int size = FileGetSize(&in);
+	if (size == 0)
+	{
+		FileClose(&in);
+		FileClose(&out);
+		return -1;
+	}
+	
+	int pos = 0, res = 1;
+	unsigned int i, chunk_size = 0x4000;
```

Comment:
Well i think maybe we should use fatfs directly for this. However forget this today.